From c5119ae34efc4ae3df6a3414bb2571aaf20b033f Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:23 -0400
Subject: [virt] kvm/vmx: fix MPX detection

Message-id: <1406040016-3289-169-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86247
O-Subject: [RHEL7 PATCH v2 168/221] KVM: vmx: fix MPX detection
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

kvm_x86_ops is still NULL at this point.  Since kvm_init_msr_list
cannot fail, it is safe to initialize it before the call.

Fixes: 93c4adc7afedf9b0ec190066d45b6d67db5270da
Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Tested-by: Jet Chen <jet.chen@intel.com>
Cc: kvm@vger.kernel.org
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 920c837785699bcc48f4a729ba9ee3492f620b95)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 919bd38..ab7261e 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -5602,9 +5602,10 @@ int kvm_arch_init(void *opaque)
   goto out_free_percpu;
 
  kvm_set_mmio_spte_mask();
- kvm_init_msr_list();
 
  kvm_x86_ops = ops;
+ kvm_init_msr_list();
+
  kvm_mmu_set_mask_ptes(PT_USER_MASK, PT_ACCESSED_MASK,
    PT_DIRTY_MASK, PT64_NX_MASK, 0);
 
-- 
1.7.1